### PR TITLE
[Feature] Add faceting settings

### LINKED ISSUE #197: Add support to the facet setting customization at the index level
⚠️ This issue is generated, which means the examples and the namings do not necessarily correspond to the language of this repository.
Also, if you are a maintainer, please add any clarification and instructions about this issue.

Related to:
- parent issue: https://github.com/meilisearch/integration-guides/issues/205
- engine issue: https://github.com/meilisearch/specifications/pull/157, https://github.com/meilisearch/meilisearch/issues/2368

----

Add a `faceting` index settings resource to manage customization of the maximum value per facet.

- Expose `faceting` object on `/indexes/:index_uid/settings` endpoints.
- Add `GET/PATCH/DELETE` - `/indexes/:index_uid/settings/faceting` endpoints.

The `faceting` object definition is made of the following properties:
- `maxValuesPerFacet` type (`int`)

More information about what each field does could be [checked in the spec.](https://github.com/meilisearch/specifications/pull/117)

Check the other customization methods for reference like: `synonyms`, `searchableAttributes`, `filterableAttributes`.

TODO:
- [ ] Add the new methods to customize `faceting`
  - [ ] GET `/indexes/:index_uid/settings/faceting`
  - [ ] PATCH `/indexes/:index_uid/settings/faceting`
  - [ ] DELETE `/indexes/:index_uid/settings/faceting`
- [ ] Add tests

### PR DESCRIPTION
**Pull request by ahmednfwela**
Fixes #197
Depends on #264 

- [x] Tests Added

### CODE REVIEW

**Comment by brunoocasali** [comment]
Can you fix the conflicts here?

**Comment by ahmednfwela** [comment]
conflicts fixed and rebased on #264

File: lib/src/faceting_settings.dart
Comment (brunoocasali) [inline lib/src/faceting_settings.dart:1]: `lib/src/faceting_settings.dart` > `lib/src/settings/faceting.dart`

File: lib/src/faceting_settings.dart [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const _defaultmaxValuesPerFacet = 100;
+
+class FacetingSettings {
```
Comment (brunoocasali) [inline lib/src/faceting_settings.dart:3]:
```suggestion
class Faceting {

```

**Comment by ahmednfwela** [comment]
Added tests, rebased on #264

**Review by brunoocasali — APPROVED** [review]
Thanks for this contribution @ahmednfwela 🎉 

bors merge

**Pull request merged**